### PR TITLE
Add support for binary serializer storage

### PR DESCRIPTION
**Pull request by sigmavirus24**
Some serializers return bytes instead of text and we need a way to
indicate that when opening the cassette file.

Related-to gh-123

--- 

cc @wimglenn for review

### CODE REVIEW

**Comment by hroncok** [comment]
The code looks reasonable. I would appreciate some kind of test tough.

**Comment by sigmavirus24** [comment]
Thanks for the prompt @hroncok. I added them in the latest commit.

File: tests/unit/test_serializers.py [hunk 1]
```diff
@@ -1,42 +1,110 @@
...
+            allow_serialization=True,
+        )
+
+    def test_serialize(self):
+        """Verify we use the right mode with open()."""
+        mopen = mock.mock_open()
+        with mock.patch('betamax.serializers.proxy.open', mopen):
+            self.proxy.serialize({'cassette': 'data'})
+
+        assert mock.call(self.cassette_path, 'wb') in mopen.mock_calls
```
Comment (hroncok) [inline tests/unit/test_serializers.py]: @sigmavirus24 Should we also check that `'w'` is used when `stored_as_binary` is `False`, or am I being to paranoid?

**Comment by wimglenn** [comment]
This "corrected file mode" method seems to be a needless complexity, I prefer my implementation where it's just using plain old class attributes on the serializer.  However, it's the same basic functionality provided, so LGTM

File: tests/unit/test_serializers.py (hunk 1 above)
Comment (sigmavirus24) [reply tests/unit/test_serializers.py]: Good point. I also simplified this test significantly.

**Comment by sigmavirus24** [comment]
> I prefer my implementation where it's just using plain old class attributes on the serializer.

There's something to be said for keeping things simpler for everyone. Your implementation is ideal if there's ever a case where a serializer wants to read text and write bytes or vice-versa, but that seems to be entirely unlikely. Simplifying what a serializer author needs to care about makes things harder to do wrong.

**Pull request merged**